core/src/main/java/org/apache/calcite/rel/rules/FilterProjectTransposeRule.java

```diff
@@ -167,43 +165,14 @@ public void onMatch(RelOptRuleCall call) {
           .replaceIfs(RelDistributionTraitDef.INSTANCE,
               () -> Collections.singletonList(
                       input.getTraitSet().getTrait(RelDistributionTraitDef.INSTANCE)));
-      newFilterRel = filter.copy(traitSet, input, simplifyFilterCondition(newCondition, call));
+      newCondition = RexUtil.removeNullabilityCast(relBuilder.getTypeFactory(), newCondition);
+      newFilterRel = filter.copy(traitSet, input, newCondition);
     } else {
-      newFilterRel =
-          relBuilder.push(project.getInput()).filter(newCondition).build();
+      newFilterRel = relBuilder.push(project.getInput())
+          .filter(newCondition)
+          .build();
     }
-
-    RelNode newProjRel =
-        copyProject
-            ? project.copy(project.getTraitSet(), newFilterRel,
-                project.getProjects(), project.getRowType())
-            : relBuilder.push(newFilterRel)
-                .project(project.getProjects(), project.getRowType().getFieldNames())
-                .build();
-
-    call.transformTo(newProjRel);
-  }
-
-  /**
-   * Simplifies the filter condition using a simplifier created by the
-   * information in the current call.
-   *
-   * <p>This method is an attempt to replicate the simplification behavior of
-   * {@link RelBuilder#filter(RexNode...)} which cannot be used in the case of
-   * copying nodes. The main difference with the behavior of that method is that
-   * it does not drop entirely the filter if the condition is always false.
-   */
-  private RexNode simplifyFilterCondition(RexNode condition, RelOptRuleCall call) {
-    final RexBuilder xBuilder = call.builder().getRexBuilder();
-    final RexExecutor executor =
-        Util.first(call.getPlanner().getContext().unwrap(RexExecutor.class),
-            Util.first(call.getPlanner().getExecutor(), RexUtil.EXECUTOR));
-    // unknownAsFalse => true since in the WHERE clause:
-    // 1>null evaluates to unknown and WHERE unknown behaves exactly like WHERE false
-    RexSimplify simplifier =
```

Comment:
Please keep these comments, they are helpful for code reading and maintaining.

Reply:
Personally i'm inclined to move the filter condition simplification out to let the `ReduceExpressionsRule.java` do its work, but with your change, this rule's behavior just changes, which means, you break the compatibility.

Reply:
Hello @danny0405, before reverting the change I'd like to ask you to agree on the comment with @zabetak since the change was made according to his [suggestion](https://github.com/apache/calcite/pull/1610#discussion_r353678039).

Reply:
Indeed it changes slightly the behavior of the rule but I think it is better not to try to do simplifications in random places around the code. I am the one who introduced the simplification inside this rule but looking back to it I think it was a mistake.

Reply:
left only ```RexUtil.removeNullabilityCast(relBuilder.getTypeFactory(), newCondition)``` and removed ```simplifyFilter``` in the rule.